packages/jupyter-ai/jupyter_ai/extension.py

```
@@ -238,41 +232,72 @@ class AiExtension(ExtensionApp):
-            chat = await self.get_chat(data["room"])
+        # log room ID
+        room_id = data["room"]
+        self.log.info(f"Connecting to a chat room with room ID: {room_id}.")
 
-            if chat is None:
-                return
+        # get YChat document associated with the room
+        ychat = await self.get_chat(data["room"])
+        if ychat is None:
+            return
 
-            # Add the bot user to the chat document awareness.
-            BOT["avatar_url"] = url_path_join(
-                self.settings.get("base_url", "/"), "api/ai/static/jupyternaut.svg"
-            )
-            if chat.awareness is not None:
-                chat.awareness.set_local_state_field("user", BOT)
+        # Add the bot user to the chat document awareness.
+        BOT["avatar_url"] = url_path_join(
+            self.settings.get("base_url", "/"), "api/ai/static/jupyternaut.svg"
+        )
+        if ychat.awareness is not None:
+            ychat.awareness.set_local_state_field("user", BOT)
+
+        # initialize chat handlers for new chat
+        self.chat_handlers_by_room[room_id] = self._init_chat_handlers(ychat)
 
-            callback = partial(self.on_change, chat)
-            chat.ymessages.observe(callback)
+        callback = partial(self.on_change, room_id)
+        ychat.ymessages.observe(callback)
 
     async def get_chat(self, room_id: str) -> Optional[YChat]:
+        """
+        Retrieves the YChat instance associated with a room ID. This method
+        is cached, i.e. successive calls with the same room ID quickly return a
+        cached value.
+
+        TODO: Determine if get_chat() should ever fail under normal usage
+        scenarios. If not, we should just raise an exception if chat is `None`,
+        and indicate the return type as just `YChat` instead of
+        `Optional[YChat]`. This will simplify the code by removing redundant
+        null checks.
+        """
```

I agree with that, we should probably raise an error, it shouldn't happen.

---

I've added this to my notes 👍